src/vs/editor/common/model/textModel.ts

```diff
@@ -979,10 +979,17 @@ export class TextModel extends OrderGuaranteeEventEmitter implements EditorCommo
 
 	private _findMatchesInLine(searchRegex:RegExp, text:string, lineNumber:number, deltaOffset:number, counter:number, result:EditorCommon.IEditorRange[], limitResultCount:number): number {
 		var m:RegExpExecArray;
+		// Reset regex to search from the beginning
+		searchRegex.lastIndex = 0;
 		do {
 			m = searchRegex.exec(text);
 			if (m) {
-				result.push(new Range(lineNumber, m.index + 1 + deltaOffset, lineNumber, m.index + 1 + m[0].length + deltaOffset));
+				var range = new Range(lineNumber, m.index + 1 + deltaOffset, lineNumber, m.index + 1 + m[0].length + deltaOffset);
```

Comment:
Because `const` and `let` are not used in the file yet `var is extensively and there's not mention in the [coding guidelines](https://github.com/Microsoft/vscode/wiki/Coding-Guidelines).